### ISSUE TITLE
WebRTC vs. WebSockets for Ubuntu ROS2 <--> iPhone

### ISSUE THREAD

**Issue by wetoo-cando** [introlab/opentera-webrtc-ros#86]
We are currently using https://github.com/christophebedard/ipad-lidar2ros to stream RGB+LiDAR data from an iPhone Pro to an Ubuntu 22.04 PC over WebSockets using [ros2-web-bridge](https://github.com/RobotWebTools/ros2-web-bridge#install).

Seems like WebRTC could be better suited for our application (streaming visual data), rather than using WebSockets that was primarily designed for transmitting text/string data using formats such as JSON.

I am guessing (hardware accelerated) codecs would also be easier to integrate with WebRTC than with WebSockets. 

Can you share your thoughts on the above issues?

**Comment by doumdi** [comment]
@wetoo-cando, yes WebRTC could be more bandwidth efficient, hardware accelerated and low latency, but it comes at the cost of :

* Compile webrtc library for ios (and all other dependencies like BoringSSL).
* Create a custom ios app to handle I/O from camera and lidar and send over WebRTC using the webrtc library.
* Use opentera-webrtc-ros nodes on the Ubuntu ROS2 system to connect the stream to the rest of ROS2 nodes.
* Install a signaling server (if your application needs to stream data from anywhere on the Internet).
* Make sure WebRTC codecs are configured properly to avoid compression artifacts on RGB and LiDAR data. You might need to use the data channel for LiDAR data, which is not really more efficient than using WebSockets anyway.

All of this is technically feasible, but would require some development efforts compared to plain WebSockets.  If you have access to the hardware from the Safari Browser on iOS, maybe the best route would be to use the WebRTC javascript interface on the browser and create a web app for streaming your RGB+LiDAR data.

Good luck with your project.

Dominic
